### PR TITLE
[GENFI] TSV improvements

### PR DESCRIPTION
**Pull request by MatthieuJoulot**
This PR aims at two things:

- [x] Correcting current issues of the `participants.tsv` and `sessions.tsv`, mainly ensuring we don't have useless information.
- [ ] Adding a `scans.tsv`, which comes in handy since it gives details about `modalities` and `runs`, as well as an information that has been required, the `path_to_the_original_dicom`.

This work will need a data CI PR, since it adds new files (`scans.tsv`) to the architecture, and should therefore break current tests.

### CODE REVIEW

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:786]: Maybe something along those lines to write rows within the loop only if the `dcm2niix` conversion was successful?

```python
if dcm2niix_success:
    row_to_write = metadata.to_csv(
        index=False, header=False, lineterminator="\t",
    ).rstrip("\t")
    with open(scans_filepath, "a") as scans_file:
         scans_file.write(row_to_write)
    if "dwi" in metadata.bids_filename and "Philips" in metadata.manufacturer:
        merge_philips_diffusion(...)
```

**Comment by MatthieuJoulot** [comment]
The data for this PR is available [here](https://github.com/aramis-lab/clinica_data_ci/pull/43)

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:801]: `scans_filepath` should already be a `Path` object :

```suggestion
            if not scans_filepath.exists():
```

File: clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py
Comment (NicolasGensollen) [inline clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:804]: I'm not sure to understand why you need to hardcode the column names and do all these steps. You basically want to write the column names (i.e. the header) only once, when creating the file. So something like this should be enough, right ?

```python
if dcm2niix_success:
    scans_filepath = ...
    row_to_write = metadata.drop(["participant_id", "session_id"]).to_csv(
        index=False,
        header=not scans_filepath.exists(),
        lineterminator="\t",
    ).rstrip("\t")
with open(scans_filepath, "a") as scans_file:
    scans_file.write(f"{row_to_write}\n")
```

Comment (NicolasGensollen) [reply clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py:804]: Hmmm, indeed it's not working as expected. Sorry about that, writing code in chat boxes without testing it is not without danger...

I had a quick look and this is the most readable way to do what you want that I could find (should work this time but to be tested in your settings...):

```python
if dcm2niix_success:
    scans_filepath = ...
    row_to_write = _serialize_row(
        metadata.drop(["participant_id", "session_id"]),
        write_column_names=not scan_filepath.exists(),
    )
with open(scans_filepath, "a") as scans_file:
    scans_file.write(f"{row_to_write}\n")


def _serialize_row(row: pd.Series, write_column_names: bool) -> str:
    row_dict = row.to_dict()
    to_write = [row_dict.keys(), row_dict.values()] if write_column_names else [row_dict.values()]
    return "\n".join([serialize_list(list(_)) for _ in to_write])


def _serialize_list(data: list, sep="\t") -> str:
    return sep.join([str(value) for value in data])
```

Comment (MatthieuJoulot) [reply clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py]: That works and it is definitely a whole lot better than what i've done

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @MatthieuJoulot !
Let's wait for the CI to finish and I'll merge if everything's green.

**Pull request merged**
